app/src/components/SectionPicker.tsx

```diff
@@ -7,46 +7,63 @@ import { getSearchByStringExample } from "@constants/frontend";
 
 interface Props {
     soc: SOC_Generic;
-    searchText: string;
-    setSearchText: (searchText: string) => void;
 }
 
 export default function SectionPicker(props: Props) {
     const [searchByString, setSearchByString] = useState<string>(
         getSearchByString(SearchBys[0]),
     );
+    const [searchText, setSearchText] = useState("");
     const searchBy = getSearchBy(searchByString);
 
     /* https://github.com/apollographql/apollo-client/issues/9583 */
     const [abortRef, setAbortRef] = useState(new AbortController());
-    const { isFetching, data: courses } = useQuery<Course[]>({
+    const {
+        isFetching,
+        data: courses,
+        refetch: fetchCourses,
+    } = useQuery<Course[]>({
         initialData: [],
-        queryKey: [props.searchText], // Re-fetch when searchText is updated
+        // queryKey: [props.searchText], // Re-fetch when searchText is updated
         queryFn: () => {
             setAbortRef(new AbortController());
             return props.soc instanceof SOC_API
-                ? props.soc.fetchSearchCourses(
-                      searchBy,
-                      props.searchText,
-                      abortRef,
-                  )
-                : props.soc.searchCourses(searchBy, props.searchText);
+                ? props.soc.fetchSearchCourses(searchBy, searchText, abortRef)
+                : props.soc.searchCourses(searchBy, searchText);
         },
-        enabled: !!props.searchText, // Prevents query when searchText is empty
+        enabled: false,
+        // enabled: !!props.searchText, // Prevents query when searchText is empty
         notifyOnChangeProps: ["data", "isFetching"], // Must re-render on isFetching change to update cursor
         refetchOnWindowFocus: false, // Turned off to prevent queries while debugging using console
         refetchOnReconnect: false, // Not needed
     });
-    console.log(isFetching ? "Fetching courses..." : "Not fetching.");
+    // console.log(isFetching ? "Fetching courses..." : "Not fetching.");
+
+    const searchCourse = () => {
```

Comment:
You should really be using `fetchCourses()`. `searchCourses()` is literally a `void` function that calls  a function with no arguments.